src/robusta/integrations/prometheus/utils.py

```diff
@@ -43,6 +89,13 @@ def check_prometheus_connection(prom: "PrometheusConnect", params: dict = None):
             # This query should return empty results, but is correct
             params={"query": "example", **params},
         )
+
+        if response.status_code == 401:
```

Comment:
I'm a little confused by this section.
We had a failure due to authorization 

We have to options:
1. Azure auth
2. Other auth

On both I think we'd like to raise an error, right? (just return nothing hides the error)

I would think you should just run `PrometheusAuthorization.request_new_token()`, then raise_for_status anyway

Regardless, this implementation suggests we'll have a Prometheus error every time the token expires, which isn't the best, right?
Can't we refresh the token before it expires?